trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/index.ts

```diff
@@ -22,12 +24,18 @@ const renderUtilisationReportReconciliationForReport = (res: Response, viewModel
 export const getUtilisationReportReconciliationByReportId = async (req: Request, res: Response) => {
   const { userToken, user } = asUserSession(req.session);
   const { reportId } = req.params;
-  const { facilityIdQuery } = req.query;
+  const { facilityIdQuery, selectedFeeRecordIds: selectedFeeRecordIdsQuery } = req.query;
 
   try {
     const facilityIdQueryAsString = facilityIdQuery ? asString(facilityIdQuery, 'facilityIdQuery') : undefined;
     const facilityIdQueryError = validateFacilityIdQuery(facilityIdQueryAsString, req.originalUrl);
-    const { errorSummary: premiumPaymentFormError, isCheckboxChecked } = getAndClearFieldsFromRedirectSessionData(req);
+
+    const { errorSummary: premiumPaymentFormError, selectedFeeRecordIds: selectedFeeRecordIdsFromSessionData } = getAndClearFieldsFromRedirectSessionData(req);
+
+    const selectedFeeRecordIdsQueryAsString = selectedFeeRecordIdsQuery ? asString(selectedFeeRecordIdsQuery, 'selectedFeeRecordIdsQuery') : undefined;
+    const selectedFeeRecordIds: Set<number> =
+      selectedFeeRecordIdsFromSessionData.size > 0 ? selectedFeeRecordIdsFromSessionData : getSelectedFeeRecordIdsFromQuery(selectedFeeRecordIdsQueryAsString);
+    const isCheckboxChecked = getIsCheckboxChecked(selectedFeeRecordIds);
```

Comment:
Are we able to fix the Codacy issues here?

Reply:
It looks as though the Codacy error on line 39 originates from the Codacy error on line 36 - but I'm unsure why this is coming up.

When we view the line in VS Code we can see that this is typed as string or undefined, not any, so I'm not sure why the errors are coming up? Any ideas? Thanks

![Screenshot 2024-08-29 at 09 17 24](https://github.com/user-attachments/assets/9f5b41b2-5010-4d27-9b9d-6e2ca6a0de7a)